sql-plugin/src/main/301until330-all/scala/com/nvidia/spark/rapids/shims/v2/Spark30Xuntil33XShims.scala

```diff
@@ -28,3 +29,8 @@ trait Spark30Xuntil33XShims extends Spark30Xuntil33XFileOptionsShims {
     GpuOverrides.neverReplaceExec[ShowCurrentNamespaceExec]("Namespace metadata operation")
   }
 }
+
+// First, Last and Collect are non-deterministic functions until Spark-3.3
```

Comment:
please clarify that these were mistakenly marked as always nondeterministic. It's not like anything changed logically in the semantics between the versions